net/sslh/src/etc/inc/plugins.inc.d/sslh.inc

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+    Copyright (C) 2021 agh1467 <7823088+agh1467@users.noreply.github.com>
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    1. Redistributions of source code must retain the above copyright notice,
+       this list of conditions and the following disclaimer.
+
+    2. Redistributions in binary form must reproduce the above copyright
+       notice, this list of conditions and the following disclaimer in the
+       documentation and/or other materials provided with the distribution.
+
+    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+ * Function to add plugin to the services page of OPNsense.
+ *
+ * This adds the service to the System: Diagnostics: Services page at
+ * http://<opnsense>/status_services.php
+ *
+ * The suffix "_services" allows this function to be picked up by
+ * etc/inc/plugins.inc:plugins_services() and be included in the
+ * $services array to be processed by status_services.php.
+ *
+ * @return array the array of attributes for this service
+*/
+function sslh_services()
+{
+
+    // Create an array to be processed by www/status_services.php
+    $service = array();
+    // Load the plugin configuraiton file as an array for use below.
+    $conf = parse_ini_file('/usr/local/opnsense/service/conf/sslh.conf', true);
```

Comment:
```suggestion
    $conf = parse_ini_file('/usr/local/opnsense/service/conf/actions.d/actions_sslh.conf', true);
```

file should be here

Reply:
Oh wait you mean to add a generic configuration... the question would be is this used somehwere else? That directory may not be the best place for it since no relation to configd then. Apologies for doing a linear review from file to file...

Reply:
Yup, that's exactly it. The intent here is to define pseudo-"super globals." I provide a broader explanation on the comment of the conf file itself.

An example of how this could be useful is, this file can be copied wholesale, change the file name, and only have to change this line, and the function name to adapt to a new service. I'd guess, it could be reduced further and eliminate the need for this file at all if this information can just be derived from the conf file. For simple services the three actions are probably enough.

Less changing values after copying/pasting the code to stand up a new plugin.

Reply:
Ok in any case the file needs to be moved. A more fitting location would be either `/usr/local/etc/inc/plugins.inc.d/sshl/sshl.conf` or `/usr/local/opnsense/data/sshl/sshl.conf`.

That being said we already have a plugin-bound meta information file that would be most fitting for these types of values, e.g.:

```
# cat /usr/local/opnsense/version/ddclient 
{
    "product_abi": "22.1",
    "product_arch": "amd64",
    "product_email": "ad@opnsense.org",
    "product_flavour": "LibreSSL",
    "product_hash": "be8192b82",
    "product_id": "os-ddclient-devel",
    "product_name": "ddclient",
    "product_version": "0.1",
    "product_website": "https://opnsense.org/"
}
```

I don't mind extending the opnsense-version utility to read manual values from it as long as we can avoid too much clutter. What I don't think is a great idea is to read this file on each file load of a model or plugin invoke, however. And I think @AdSchellevis will agree. For configuration purposes it's fine but for runtime always bouncing through the file or shell command is a performance killer.

Reply:
Since the practicality for this isn't represented in this small plugin I've decided to remove it and replace with hard coded strings.